classes/ActionScheduler_WPCommentCleaner.php

```diff
@@ -66,13 +66,17 @@ public static function has_logs() {
 	 * Attached to the migration complete hook 'action_scheduler/migration_complete'.
 	 */
 	public static function maybe_schedule_cleanup() {
+		$has_logs = 'no';
+
 		if ( (bool) get_comments( array( 'type' => ActionScheduler_wpCommentLogger::TYPE, 'number' => 1, 'fields' => 'ids' ) ) ) {
-			update_option( self::$has_logs_option_key, 'yes' );
+			$has_logs = 'yes';
 
 			if ( ! as_next_scheduled_action( self::$cleanup_hook ) ) {
 				as_schedule_single_action( gmdate( 'U' ) + ( 6 * MONTH_IN_SECONDS ), self::$cleanup_hook );
 			}
 		}
+
+		update_option( self::$has_logs_option_key, $has_logs );
```

Comment:
```suggestion
		update_option( self::$has_logs_option_key, $has_logs, true );
```

Since this option gets queried on every page load, I'm thinking the autoload should be explicitly set to true. [Currently, WP heuristics would presumably determine this to be true anyway](https://make.wordpress.org/core/2024/06/18/options-api-disabling-autoload-for-large-options/), but that could change in the future.